src/serial-provision-dialog.ts

```diff
@@ -180,95 +202,130 @@ class SerialProvisionDialog extends LitElement {
-                    ? null
-                    : this._ssids![index].name;
-              }}
-              @closed=${(ev: Event) => ev.stopPropagation()}
-            >
-              ${this._ssids!.map(
-                (info, idx) => html`
-                  <is-list-item .selected=${selectedSsid === info} value=${idx}>
-                    ${info.name}
-                  </is-list-item>
-                `
-              )}
-              <is-list-item .selected=${!selectedSsid} value="-1">
-                Join other…
-              </is-list-item>
-            </is-select>
-            <ewt-icon-button @click=${this._updateSsids}>
-              ${refreshIcon}
-            </ewt-icon-button>
+            <div class="network-select">
+              <md-outlined-select
+                name="ssid_select"
+                required
+                label="Network"
+                @change=${(ev: Event) => {
+                  const index = (ev.target as MdOutlinedSelect).selectedIndex;
+                  // The "Join Other" item is always the last item.
+                  this._selectedSsid =
+                    index === this._ssids!.length
+                      ? null
+                      : this._ssids![index].name;
+                }}
+                @closed=${(ev: Event) => ev.stopPropagation()}
+              >
+                ${this._ssids!.map(
+                  (info, idx) => html`
+                    <md-select-option
+                      .selected=${selectedSsid === info}
+                      value=${idx}
+                    >
+                      <md-icon
+                        slot="start"
+                        class=${getSignalStrengthClass(info.rssi)}
+                        >${getWifiIconName(info.rssi)}</md-icon
+                      >
+                      <span slot="headline">${info.name}</span>
+                      <span slot="end" class="network-details">
+                        <span class="signal-strength">${info.rssi}dB</span>
+                        <md-icon
+                          class="lock-icon ${info.secured
+                            ? "lock-secured"
+                            : "lock-unsecured"}"
+                          >${info.secured ? "lock" : "lock_open"}</md-icon
+                        >
+                      </span>
+                    </md-select-option>
+                  `,
+                )}
+                <md-select-option .selected=${!selectedSsid} value="-1">
+                  Join other…
+                </md-select-option>
+              </md-outlined-select>
+
+              <md-outlined-icon-button @click=${this._updateSsids} data-refresh>
+                <md-icon>refresh</md-icon>
+              </md-outlined-icon-button>
+            </div>
           `
         : ""}
       ${
         // Show input box if command not supported or "Join Other" selected
-        !selectedSsid
+        this._ssids === null || this._selectedSsid === null
```

Comment:
We will now show a text box when the SSIDs are still loading? Can we just show a loading spinner and wait up to 5s for first SSIDs to come in?

Reply:
This is for when Scanning Wifi is not supported as well. I can also fix it to not show until there are at least some SSIDs retrieved when scan wifi is supported. Don't need the timeout.